exporter/sentryexporter/sentry_exporter.go

```diff
@@ -56,26 +56,106 @@ type SentryExporter struct {
 	DSN string
 }
 
+// IDMap maps a span_id to a root span span_id.
+type IDMap map[string]string
+
+// SpanStore stores a root span and it's child spans.
+type SpanStore struct {
+	rootSpan   *SentrySpan
+	childSpans []*SentrySpan
+}
+
+// SSMap maps a root span span_id to a SpanStore.
+type SSMap map[string]*SpanStore
+
 // TODO: Add to function
 func (s *SentryExporter) pushTraceData(ctx context.Context, td pdata.Traces) (droppedSpans int, err error) {
+	// For a ResourceSpan, InstrumentationLibrarySpan and Span struct if IsNil() is "true", all other methods will cause a runtime error.
+	resourceSpans := td.ResourceSpans()
+	if resourceSpans.Len() == 0 {
+		return 0, nil
+	}
+
+	orphanSpans := make([]*SentrySpan, 0, td.SpanCount())
+
+	// Maps all child span ids to their root span.
+	idMap := make(IDMap)
+	// Maps root span id to a root span and it's child span.
+	ssMap := make(SSMap)
+
+	for i := 0; i < resourceSpans.Len(); i++ {
+		rs := resourceSpans.At(i)
+		if rs.IsNil() {
+			continue
+		}
+
+		ilss := rs.InstrumentationLibrarySpans()
+		for j := 0; j < ilss.Len(); j++ {
+			ils := ilss.At(j)
+			if ils.IsNil() {
+				continue
+			}
+
+			spans := ils.Spans()
+			for k := 0; k < spans.Len(); k++ {
+				otelSpan := spans.At(k)
+				if otelSpan.IsNil() {
+					continue
+				}
+
+				sentrySpan := convertToSentrySpan(otelSpan)
+
+				if sentrySpan.IsRootSpan() {
+					// Add root span to span store map
+					ssMap[sentrySpan.SpanID] = &SpanStore{
+						rootSpan:   sentrySpan,
+						childSpans: make([]*SentrySpan, 0),
+					}
+
+					idMap[sentrySpan.SpanID] = sentrySpan.SpanID
+				} else {
+					if rootSpanID, ok := idMap[sentrySpan.ParentSpanID]; ok {
+						idMap[sentrySpan.SpanID] = rootSpanID
+						ssMap[rootSpanID].childSpans = append(ssMap[rootSpanID].childSpans, sentrySpan)
```

Comment:
Do the two calls to `ssMap[rootSpanID].childSpans` mean two lookups in the hash table? 
And if so would assigning the return to a local var `childSpans` avoid the second lookup?

```suggestion
						ssMap[rootSpanID].childSpans = append(ssMap[rootSpanID].childSpans, sentrySpan)
```